### PR TITLE
Dialog: Replies loader

### PR DESCRIPTION
**Pull request by siddharthkp**
## What steps did you take to test this? This is required before we can merge, make sure to test the flow you've updated.

1. Step A
2. Step B
3. Step C

## Checklist

- [ ] Documentation
- [ ] Testing 
- [ ] Ready to be merged
      
- [ ] Added myself to contributors table

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/3781/commits/0152189de71f946bcc71b80fed7cfbbb383ebfc0 is at https://pr3781.build.csb.dev/s/new.

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Comments/Dialog/index.tsx [hunk 1]
```diff
@@ -331,13 +346,49 @@ const CommentBody = ({ comment, editing, setEditing }) => {
...
 
-const Replies = ({ replies }) => (
-  <>
-    {replies.map(reply =>
-      reply ? <Reply reply={reply} key={reply.id} /> : 'Loading...'
-    )}
-  </>
-);
+const Replies = ({ replies, repliesRenderedCallback }) => {
+  const [repliesLoaded, setRepliesLoaded] = React.useState(!!replies[0]);
```
Comment (christianalfoni) [inline packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Comments/Dialog/index.tsx]: Do we need this? Can just use `const repliesLoaded = !!replies[0]`?

Comment (siddharthkp) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Comments/Dialog/index.tsx]: Ah, we might not need it anymore, will check!

Comment (siddharthkp) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Comments/Dialog/index.tsx]: Done!

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Comments/Dialog/index.tsx [hunk 2]
```diff
@@ -331,13 +346,45 @@ const CommentBody = ({ comment, editing, setEditing }) => {
...
-const Replies = ({ replies }) => (
-  <>
-    {replies.map(reply =>
-      reply ? <Reply reply={reply} key={reply.id} /> : 'Loading...'
-    )}
-  </>
-);
+const Replies = ({ replies, repliesRenderedCallback }) => {
+  const [isAnimating, setAnimating] = React.useState(true);
+  const repliesLoaded = !!replies[0];
```
Comment (MichaelDeBoey) [inline packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Comments/Dialog/index.tsx]:
```suggestion
  const repliesLoaded = Boolean(replies[0]);
```

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Comments/Dialog/index.tsx [hunk 3]
```diff
@@ -331,13 +346,45 @@ const CommentBody = ({ comment, editing, setEditing }) => {
...
+
+  /** Wait another <delay>ms after the dialog has transitioned into view */
+  const delay = DIALOG_TRANSITION_DURATION + REPLY_TRANSITION_DELAY;
+  const REPLY_TRANSITION_DURATION = 0.25;
+  const SKELETON_HEIGHT = 146;
+
+  React.useEffect(() => {
+    if (repliesLoaded && !isAnimating) repliesRenderedCallback();
```
Comment (MichaelDeBoey) [inline packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Comments/Dialog/index.tsx]:
```suggestion
    if (repliesLoaded && !isAnimating) {
      repliesRenderedCallback();
    }
```

File: packages/components/src/components/SkeletonText/index.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import styled, { css as styledcss, keyframes } from 'styled-components';
+import Color from 'color';
+import { Element } from '../Element';
+
+// export interface ITextProps extends React.HTMLAttributes<HTMLSpanElement> {}
+
```
Comment (MichaelDeBoey) [inline packages/components/src/components/SkeletonText/index.tsx]:
```suggestion
```

**Pull request merged**